### PR TITLE
Add Dart Examples for Fiat Currencies, Connecting to an LSP & channel opening fees calculation

### PR DESCRIPTION
**Pull request by erdemyerebasmaz**

### CODE REVIEW

**Review by roeierez — CHANGES_REQUESTED** [review]
Worth going over the English. Maybe @dangeross can help here?

File: src/guide/connecting_lsp.md [hunk 1]
```diff
@@ -3,6 +3,29 @@
 In order to list all available lsp you may connect to you may do the following.
 
 <custom-tabs category="lang">
+<div slot="title">Dart</div>
+<section>
+
+```dart
```
[the remark this answers is not in the document]
Comment (dangeross) [reply src/guide/connecting_lsp.md:9]: Something like:

> Based on the API key provided to the Breez SDK, a default LSP is selected for your node to provide liquidity to it. You get the information about the select LSP you can do the following:

File: src/guide/connecting_lsp.md [hunk 2]
```diff
@@ -3,6 +3,29 @@
 In order to list all available lsp you may connect to you may do the following.
 
 <custom-tabs category="lang">
+<div slot="title">Dart</div>
+<section>
+
```
Comment (erdemyerebasmaz) [inline src/guide/connecting_lsp.md:8]:
```suggestion

Based on the API key provided to the Breez SDK, a default LSP is selected for your node to provide liquidity to it. To get the information about the selected LSP you can do the following:

```

**Comment by erdemyerebasmaz** [comment]
@roeierez @dangeross A paragraph for fee calculation would be great.
I don't know how simple we want to keep this, I presume it wouldn't involve details about HTLC's and what not.

**Comment by roeierez** [comment]
> @roeierez @dangeross A paragraph for fee calculation would be great. I don't know how simple we want to keep this, I presume it wouldn't involve details about HTLC's and what not.

No need to go into technical details at all. The user only needs to understand:
1. How to detect if open channel fees are needed
2. How to calculate the fee for a specific amount
3. Code sample for 2

**Comment by erdemyerebasmaz** [comment]
> > @roeierez @dangeross A paragraph for fee calculation would be great. I don't know how simple we want to keep this, I presume it wouldn't involve details about HTLC's and what not.
> 
> No need to go into technical details at all. The user only needs to understand:
> 
> 1. How to detect if open channel fees are needed
> 2. How to calculate the fee for a specific amount
> 3. Code sample for 2

Addressed here: 39682c55e43622e2a1b6688a7bbde9adc17cfaa7
Feel free to change wording.

File: src/guide/receive_onchain.md [hunk 3]
```diff
@@ -283,3 +283,51 @@ result, err := sdkServices.Refund(refundable.BitcoinAddress, destinationAddress,
...
 </section>
 </custom-tabs>
+
+# Calculating fees
+
+<custom-tabs category="lang">
+<div slot="title">Dart</div>
+<section>
+
+To calculate fees, you'll need to detect if channel opening fees are needed first and then calculate the fee for the amount you'll be receiving.
```
Comment (dangeross) [inline src/guide/receive_onchain.md:293]: When the amount to be received exceeds the inbound liquidity of the node, a new channel will be opened by the LSP in order for the node to receive it. This can checked by retrieving the NodeState from the SDK and comparing the inbound liquidity to the amount to be received. If the amount is greater or equal to the inbound liquidity, a new channel opening is required.

To calculate the fees for a channel being opened by the LSP:

**Comment by dangeross** [comment]
Feel free to use/change what you want. Maybe its too wordy now 🙂

**Comment by ubbabeck** [comment]
Accidentally closed this, my bad,, but please feel free to continue the discussion in this pr or here https://github.com/breez/breez-sdk-docs/pull/18.
